packages/explat-client/src/internal/test/requests.ts

```diff
@@ -0,0 +1,233 @@
+/**
+ * External dependencies
+ */
+// This is required to fix the "regeneratorRuntime is not defined" error
+import '@automattic/calypso-polyfills';
+
+/**
+ * Internal dependencies
+ */
+import * as Timing from '../timing';
+import * as Requests from '../requests';
+import type { Config, ExperimentAssignment } from '../../types';
+import { delayedValue, validExperimentAssignment } from '../test-common';
+import * as ExperimentAssignments from '../experiment-assignments';
+
+const spiedMonotonicNow = jest.spyOn( Timing, 'monotonicNow' );
+
+const mockedFetchExperimentAssignment = jest.fn();
+const mockedGetAnonId = jest.fn();
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore; Not using the full Config
+const mockedConfig: Config = {
+	fetchExperimentAssignment: mockedFetchExperimentAssignment,
+	getAnonId: mockedGetAnonId,
+};
+
+function mockFetchExperimentAssignmentToMatchExperimentAssignment(
+	experimentAssignment: ExperimentAssignment
+) {
+	spiedMonotonicNow.mockImplementationOnce( () => experimentAssignment.retrievedTimestamp );
+	mockedFetchExperimentAssignment.mockImplementationOnce( () =>
+		delayedValue(
+			{
+				ttl: experimentAssignment.ttl,
+				variations: {
+					[ experimentAssignment.experimentName ]: experimentAssignment.variationName,
+				},
+			},
+			1
+		)
+	);
+}
+
+describe( 'fetchExperimentAssignment', () => {
+	it( 'should successfully fetch and return a well formed response', async () => {
+		mockedGetAnonId.mockImplementationOnce( () => delayedValue( 'asdf', 1 ) );
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( validExperimentAssignment );
+		const experimentAssignmentWithAnonId = await Requests.fetchExperimentAssignment(
+			mockedConfig,
+			validExperimentAssignment.experimentName
+		);
+		expect( experimentAssignmentWithAnonId ).toEqual( validExperimentAssignment );
+		expect( mockedFetchExperimentAssignment.mock.calls ).toEqual( [
+			[
+				{
+					anonId: 'asdf',
+					experimentName: 'experiment_name_a',
+				},
+			],
+		] );
+
+		mockedFetchExperimentAssignment.mockReset();
+		mockedGetAnonId.mockImplementationOnce( () => delayedValue( null, 1 ) );
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( validExperimentAssignment );
+		const experimentAssignmentWithoutAnonId = await Requests.fetchExperimentAssignment(
+			mockedConfig,
+			validExperimentAssignment.experimentName
+		);
+
+		expect( experimentAssignmentWithoutAnonId ).toEqual( validExperimentAssignment );
+		expect( mockedFetchExperimentAssignment.mock.calls ).toEqual( [
+			[
+				{
+					anonId: null,
+					experimentName: 'experiment_name_a',
+				},
+			],
+		] );
```

Comment:
It'd be nice to split this into a separate `it()` scenario since it's pretty much independent from the first scenario. That is, have `it( 'should successfully fetch and return a well formed response with an anon ID' )` and `it( 'should successfully fetch and return a well formed response without an anon ID' )`.